File: client/src/app/_dtos/finance/voucherWithNewAttachmentDto.ts
import { IVoucher } from "src/app/_models/finance/voucher";
import { IVoucherAttachment } from "src/app/_models/finance/voucherAttachment";

export interface IVoucherWithNewAttachmentDto
{
    voucher: IVoucher;
    newAttachments: IVoucherAttachment[];
}

File: api/DTOs/Finance/VoucherWithNewAttachmentDto.cs
using api.Entities.Finance;

namespace api.DTOs.Finance
{
    public class VoucherWithNewAttachmentDto
    {
        public FinanceVoucher Voucher { get; set; }
        public ICollection<VoucherAttachment> NewAttachments { get; set; }
    }
}